core/src/test/scala/unit/kafka/tools/ConsoleProducerTest.scala

```diff
@@ -40,6 +40,8 @@ class ConsoleProducerTest {
     "t3"
   )
 
+  val invalidParseArgs: Array[String] = validArgs :+ "--key-serializer" :+ "org.apache.kafka.common.serialization.StringSerializer"
```

Comment:
@aishraj I think there should be a test case for value-serializer as well. This could have caught the error above.